hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSource.java

```diff
@@ -380,7 +380,8 @@ private List<MergeOnReadInputSplit> buildFileIndex() {
             .path(FilePathUtils.toFlinkPath(path))
             .rowType(this.tableRowType)
             .maxCompactionMemoryInBytes(maxCompactionMemoryInBytes)
-            .requiredPartitions(getRequiredPartitionPaths()).build();
+            .requiredPartitions(getRequiredPartitionPaths())
+            .skipClustering(false).build();
```

Comment:
The `skipClustering` is false at default, therefore this doesn't need to set to false, and could set to `conf.getBoolean(FlinkOptions.READ_STREAMING_SKIP_CLUSTERING)` or no set.

Reply:
removed the line

Reply:
Question, is it safer to hardcode a false into this under `BatchMode`? 

IIUC, when reading in `batchMode`, `skip_clustering` should always be `false` and should not be configurable. 

This is so as the reader will have an empty "state", i.e. no prior reads performed. As such, there will be no "previous" records that have been read. So, reading of reading of `replacecommits` must always be performed to ensure correctness.

I suggest adding a comment to remind anyone modifying this part of the code in the future of **WHY** this `skip_clustering`  must be **false** here.